### PR TITLE
Manage python type content

### PR DESCRIPTION
**Pull request by maximiliankolb**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.7/Katello 4.9 (planned Satellite 6.14)
* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (Satellite 6.13)

### CODE REVIEW

File: guides/common/modules/con_managing-custom-python-type-content.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[id="Managing_{custompythontypeid}_Type_Content_{context}"]
+= Managing {custompythontypetitle} Content
+
+You can use {Project} to manage {custompythontype} repositories.
+To achieve this, {customproduct}s in {ProjectName} include repositories for {custompythontype}.
+This provides a generic method to incorporate python packages in a product.
+
+You can upload python packages to {custompythonrepos} and synchronize files from an upstream {ProjectServer}.
+When you add python packages to a {custompythontype} repository, you can use the normal {Project} management functions such as adding a specific version to a Content View to provide version control and making the repository available on various {SmartProxyServers}.
+Clients must download the files over HTTP or HTTPS using `curl -O`.
```
Comment (adamlazik1) [inline guides/common/modules/con_managing-custom-python-type-content.adoc:10]:
```suggestion
You must download the files on clients over HTTP or HTTPS using `curl -O`.
```

File: guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+You can download files to a client over HTTPS using `curl -O`, and optionally over HTTP if the *Unprotected* option for repositories is selected.
+
+.Prerequisites
+* You have a {custompythontype} repository.
+* You know the name of the python package you want to download to clients from the {custompythontype} repository.
+* To download the python packages over HTTPS to your managed hosts, you require:
++
+. The `katello-server-ca.crt`.
+For more information, see {AdministeringDocURL}Importing_the_Katello_Root_CA_Certificate_admin[Importing the Katello Root CA Certificate] in _{AdministeringDocTitle}_.
+. An Organization Debug Certificate.
```
Comment (adamlazik1) [inline guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc:13]:
```suggestion
** The `katello-server-ca.crt`.
For more information, see {AdministeringDocURL}Importing_the_Katello_Root_CA_Certificate_admin[Importing the Katello Root CA Certificate] in _{AdministeringDocTitle}_.
** An Organization Debug Certificate.
```

File: guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+For more information, see {ManagingOrganizationsLocationsDocURL}Creating_an_Organization_Debug_Certificate_managing-organizations-locations[Creating an Organization Debug Certificate] in _{ManagingOrganizationsLocationsDocTitle}_.
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Content* > *Products*.
+. Select a {customproduct} by name.
+. Select a `python` type repository by name.
+. Optional: Select the *Unprotected* checkbox to access the repository published through HTTP.
+. Copy the *Published At* URL.
```
Comment (adamlazik1) [inline guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc:21]: This is missing the final step which the CLI procedure has. IMHO it's fine if you copy and paste but you can also create a snippet. Your call.

File: guides/common/modules/proc_installing-python-packages-from-server.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+.Procedure
+* On your managed hosts, set the URL to {ProjectServer} to install a python package:
++
+[options="nowrap" subs="+quotes,verbatim,attributes"]
+----
+$ pip install --index-url https://{foreman-example-com}/pulp/content/_My_Organization_Label_/Library/custom/_My_{customproductid}/_My_{custompythonrepoid}/simple/ _My_Python_Package_
+----
++
+If your managed host does not trust the certificate of your {ProjectServer}, use `--trusted-host {foreman-example-com}`.
```
Comment (adamlazik1) [inline guides/common/modules/proc_installing-python-packages-from-server.adoc:17]:
```suggestion
If your managed host does not trust the certificate of your {ProjectServer}, add the `--trusted-host {foreman-example-com}` option.
```

File: guides/common/modules/proc_synchronizing-custom-python-repositories.adoc [hunk 5]
```diff
@@ -0,0 +1,37 @@
+[id="Synchronizing_{custompythonreposid}_{context}"]
+= Synchronizing {custompythonrepostitle}
+
+On {Project}, you can synchronize your {custompythonrepos} from any mirror and other {Project} instances.
+Your can view the python packages in the {ProjectWebUI} at *Content* > *Other Content Types*.
```
Comment (adamlazik1) [inline guides/common/modules/proc_synchronizing-custom-python-repositories.adoc:5]:
```suggestion
You can view the python packages in the {ProjectWebUI} at *Content* > *Other Content Types*.
```

File: guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+.Procedure
+. In the {ProjectWebUI}, navigate to *Content* > *Products*.
+. Select a {customproduct} by name.
+. Select a `python` type repository by name.
+. Click *Browse* to search and select the file you want to upload.
++
+You can upload `.whl` packages and `.tar.gz` archives that contain python packages.
+. Click *Upload* to upload the selected file to {ProjectServer}.
+. Visit the *Published At* URL to see the file.
```
Comment (adamlazik1) [inline guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc:14]: This is potentially a verification step.

File: guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+.CLI procedure
+[options="nowrap" subs="+quotes"]
+----
+# hammer repository upload-content \
+--id _My_Repository_ID_ \
+--organization-id _My_Organization_ID_ \
+--path _My_File_
```
Comment (adamlazik1) [inline guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc:22]:
```suggestion
--path _Path_To_My_File_
```

File: guides/common/attributes-base.adoc [hunk 8]
```diff
@@ -47,9 +47,18 @@
...
 :customrepotitle: Repository
 :customrepoid: Repository
+:custompythonrepo: python repository
+:custompythonrepoid: Python_Repository
+:custompythonrepos: python repositories
+:custompythonreposid: Python_Repositories
+:custompythonrepostitle: Python Repositories
+:custompythontype: python type
+:custompythontypeid: Python_Type
+:custompythontypetitle: Python Type
```
Comment (adamlazik1) [inline guides/common/attributes-base.adoc:61]: Is this something that can really differ accross builds?

**Review by adamlazik1 — COMMENTED** [review]
Provided some comments.

File: guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc (hunk 3 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc:20]: I've created a snippet for this. Do you think it makes sense to include the snippet twice; for ".Produce" and ".CLI procedure"?

File: guides/common/modules/proc_installing-python-packages-from-server.adoc (hunk 4 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_installing-python-packages-from-server.adoc]: Applied.

File: guides/common/modules/proc_synchronizing-custom-python-repositories.adoc (hunk 5 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_synchronizing-custom-python-repositories.adoc]: Applied.

File: guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc (hunk 6 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc]: Moved to ".Verification". Is this supposed to be before or after the CLI procedure?

**Review by maximiliankolb — COMMENTED** [review]
Thanks Adam.

Rebased to master and applied all feedback to a second commit.

File: guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc (hunk 3 above)
Comment (adamlazik1) [reply guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc:20]: Yes, I would include it twice just for completeness.

File: guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc (hunk 6 above)
Comment (adamlazik1) [reply guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc]: After.

File: guides/common/modules/snip_downloading-file-from-server.adoc [hunk 9]
```diff
@@ -0,0 +1,21 @@
+. On your client, download the file from {ProjectServer}:
++
+For HTTPS:
```
Comment (adamlazik1) [inline guides/common/modules/snip_downloading-file-from-server.adoc:3]:
```suggestion
* For HTTPS:
```

File: guides/common/modules/snip_downloading-file-from-server.adoc [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+[options="nowrap" subs="+quotes,attributes"]
+----
+# curl \
+--cacert _./_katello-server-ca.crt_ \
+--cert _./_My_Organization_key-cert.pem_ \
+--remote-name \
+https://_{foreman-example-com}_/pulp/content/_My_Organization_Label_/Library/custom/_My_Product_Label_/_My_Repository_Label_/_My_File_
+----
++
+For HTTP:
```
Comment (adamlazik1) [inline guides/common/modules/snip_downloading-file-from-server.adoc:14]:
```suggestion
* For HTTP:
```

**Review by adamlazik1 — COMMENTED** [review]
Looks good! Left two minor nitpicks.

File: guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc (hunk 3 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_downloading-python-packages-to-a-host-from-a-custom-python-repository.adoc:20]: OK, no change required then.

File: guides/common/modules/snip_downloading-file-from-server.adoc (hunk 9 above)
Comment (maximiliankolb) [reply guides/common/modules/snip_downloading-file-from-server.adoc]: Applied.

File: guides/common/modules/snip_downloading-file-from-server.adoc (hunk 10 above)
Comment (maximiliankolb) [reply guides/common/modules/snip_downloading-file-from-server.adoc]: Applied.

**Review by maximiliankolb — COMMENTED** [review]
Rebased to "master" and applied all feedback.

File: guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc (hunk 6 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_uploading-files-to-a-custom-python-repository.adoc]: Moved!

**Review by adamlazik1 — APPROVED** [review]
LGTM

File: guides/common/modules/proc_downloading-files-to-a-host-from-a-custom-file-repository.adoc [hunk 11]
```diff
@@ -24,7 +23,8 @@ endif::[]
 . Select a {customproduct} by name.
 . Select a file type repository by name.
 . Ensure to select the *Unprotected* checkbox to access the repository published through HTTP.
-. Copy the URL where the repository is published.
+. Copy the *Published At* URL.
+include::snip_downloading-file-from-server.adoc[]
```
Comment (Lennonka) [inline guides/common/modules/proc_downloading-files-to-a-host-from-a-custom-file-repository.adoc:27]:
```suggestion
include::snip_step-downloading-file-from-server.adoc[]
```
I'd add the "snippet type" to the file name.

File: guides/common/modules/proc_synchronizing-custom-python-repositories.adoc [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+. From the *Type* list, select *python*.
+. In the *Upstream URL* field, enter the URL for the upstream content source, for example `\https://pypi.org`.
+. Optional: In the *Upstream Username* field, enter the user name for the upstream repository if required for authentication.
+Clear this field if the repository does not require authentication.
+. Optional: In the *Upstream Password* field, enter the corresponding password for the upstream repository.
+Clear this field if the repository does not require authentication.
+. Optional: In the *Upstream Authentication Token* field, provide the token of the upstream repository user for authentication.
+Leave this field empty if the repository does not require authentication.
+. Optional: In the *Excludes* field, enter a list of python packages separated by newlines to exclude from synchronizing to {Project}.
+. Optional: In the *Includes* field, enter a list of python packages separated by newlines to limit the synchronized packages.
```
Comment (Lennonka) [inline guides/common/modules/proc_synchronizing-custom-python-repositories.adoc:24]:
```suggestion
. Optional: In the *Includes* field, enter a list of python packages separated by newlines to limit the synchronization only to the included packages.
```
To be more specific, if this is the case.

**Review by Lennonka — CHANGES_REQUESTED** [review]
A couple of nitpicks, I'll do another round of review later.

File: guides/doc-Managing_Content/master.adoc [hunk 13]
```diff
@@ -55,6 +55,10 @@ include::common/assembly_managing-isos-and-files.adoc[leveloffset=+1]
 
 include::common/assembly_managing-custom-file-type-content.adoc[leveloffset=+1]
 
+ifndef::satellite[]
+include::common/assembly_managing-custom-python-type-content.adoc[leveloffset=+1]
```
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/doc-Managing_Content/master.adoc:59]: FYI, the "custom" file type is called "custom File Type" only in Satellite and just "File Type" in everything else.
Therefore, we should go here just with "Python Type" as we don't need to modify the terms for any build.

File: guides/common/attributes-base.adoc [hunk 14]
```diff
@@ -48,9 +48,18 @@
...
```
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/common/attributes-base.adoc:62]: Yes, then intention was to use attributes if Satellite decides to use "custom python content" in the future. Yay or nay? I am OK either way.

File: guides/common/modules/proc_downloading-files-to-a-host-from-a-custom-file-repository.adoc (hunk 11 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_downloading-files-to-a-host-from-a-custom-file-repository.adoc]: Renamed.

File: guides/common/modules/proc_synchronizing-custom-python-repositories.adoc (hunk 12 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_synchronizing-custom-python-repositories.adoc]: Applied.

File: guides/doc-Managing_Content/master.adoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/doc-Managing_Content/master.adoc]: Removed the word "custom".

File: guides/common/attributes-base.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (maximiliankolb) [reply guides/common/attributes-base.adoc]: After renaming "custom python type content" to "python type content", I've removed the attributes in the last commit. Let me know what you think.

File: guides/common/modules/proc_uploading-files-to-a-python-repository.adoc [hunk 15]
```diff
@@ -0,0 +1,28 @@
+[id="Uploading_Files_To_a_Python_Type_Repository_{context}"]
+= Uploading Files To a Python Type Repository
+
+Use this procedure to upload files to a python type repository.
```
Comment (Lennonka) [inline guides/common/modules/proc_uploading-files-to-a-python-repository.adoc:4]:
```suggestion
You can upload local files to a python type repository.
```
or
```suggestion
You can upload files from your local machine to a python type repository.
```

File: guides/common/attributes-base.adoc (hunk 14 above)
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/common/attributes-base.adoc]: Definitely nay for "potential futures".

It looks great now!

**Review by Lennonka — APPROVED** [review]
A detail, otherwise LGTM!

File: guides/common/modules/proc_uploading-files-to-a-python-repository.adoc (hunk 15 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_uploading-files-to-a-python-repository.adoc]: Applied.

**Review by maximiliankolb — COMMENTED** [review]
Rebased to master and applied the last suggestion by Lena.

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: :heavy_check_mark: 
   14af2a86d..b9e14e88f  3.7 -> 3.7
   0a7cb719a..05c86c81e  3.6 -> 3.6
   2ef7165a0..975eb358b  3.5 -> 3.5

**Pull request merged**